scalr/resource_scalr_provider_configuration_default.go

```diff
@@ -0,0 +1,208 @@
+		},
+
+		Schema: map[string]*schema.Schema{
+			"provider_configuration_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"environment_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceScalrProviderConfigurationDefaultImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	scalrClient := meta.(*scalr.Client)
+
+	id := d.Id()
+
+	providerConfiguration, environment, err := getPCDLinkedResources(ctx, id, scalrClient)
+	if err != nil {
+		if errors.Is(err, scalr.ErrResourceNotFound) {
+			return nil, fmt.Errorf("provider configuration default %s not found", id)
+		}
+		return nil, fmt.Errorf("error retrieving provider configuration default %s: %v", id, err)
+	}
+
+	_ = d.Set("provider_configuration_id", providerConfiguration.ID)
+	_ = d.Set("environment_id", environment.ID)
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func resourceScalrProviderConfigurationDefaultCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	scalrClient := meta.(*scalr.Client)
+
+	providerConfigurationID := d.Get("provider_configuration_id").(string)
+	environmentID := d.Get("environment_id").(string)
+	id := fmt.Sprintf("%s/%s", environmentID, providerConfigurationID)
+
+	if !validStringID(&environmentID) {
+		return diag.Errorf("invalid value for environment ID: %q", environmentID)
+	}
+
+	if !validStringID(&providerConfigurationID) {
+		return diag.Errorf("invalid value for provider configuration ID: %q", providerConfigurationID)
+	}
+
+	environment, err := scalrClient.Environments.Read(ctx, environmentID)
+	if err != nil {
+		if errors.Is(err, scalr.ErrResourceNotFound) {
+			return diag.Errorf("Environment %q not found", environmentID)
+		}
+		return diag.Errorf("error retrieving environment %s: %v", environmentID, err)
+	}
+
+	providerConfiguration, err := scalrClient.ProviderConfigurations.Read(ctx, providerConfigurationID)
+	if err != nil {
+		if errors.Is(err, scalr.ErrResourceNotFound) {
+			return diag.Errorf("Provider configuration %q not found", providerConfigurationID)
+		}
+		return diag.Errorf("Error retrieving provider configuration %s: %v", providerConfigurationID, err)
+	}
+
+	for _, pc := range environment.DefaultProviderConfigurations {
+		if pc.ID == providerConfigurationID {
+			return diag.Errorf("Provider configuration %q is already in environment %q default provider configuration", providerConfigurationID, environmentID)
+		}
+	}
+
+	environment.DefaultProviderConfigurations = append(environment.DefaultProviderConfigurations, &scalr.ProviderConfiguration{ID: providerConfiguration.ID})
+	updateOpts := scalr.EnvironmentUpdateOptions{
+		DefaultProviderConfigurations: environment.DefaultProviderConfigurations,
+	}
+	_, err = scalrClient.Environments.Update(ctx, environment.ID, updateOpts)
```

Comment:
This won't work as expected. Only the last one resource to create will succeed; others will produce inconsistent results on read (parallelism again).